tracer/src/Datadog.Trace/SpanContext.cs

```diff
@@ -257,6 +257,11 @@ internal string Origin
 
         internal PathwayContext? PathwayContext { get; private set; }
 
+        /// <summary>
+        ///  Gets or sets a value indicating whether this <see cref="SpanContext"/> was propagated from a remote parent.
+        /// </summary>
+        internal bool IsRemote { get; set; }
```

Comment:
If `IsRemote` should never change once the `SpanContext` is created, consider setting `IsRemote` through the constructor only and make this get-only.